pkg/webhook/admission/pod/mutator.go

```diff
@@ -116,15 +123,15 @@ func (mutator *Mutator) mutate(pod *v1.Pod, configMap *v1.ConfigMap) error {
 		return err
 	}
 
-	mutators := []func(pod *v1.Pod) error{
+	mutators := []func(pod *v1.Pod, targetNs *v1.Namespace) error{
```

Comment:
Instead of changing the signature of every injector by a namespace argument that mostly is ignored except for one injector, I would instead use a closure for the one injector that needs a namespace and leave the signature of the others unchanged, to avoid unnecessary churn. Especially that you already know that namespace at this point

E.g. 

```
mutators := []func(pod *v1.Pod) error {
                InjectGKEAcceleratorSelector,
		func(pod *v1.Pod) error { 
                     return storageInitializer.InjectStorageInitializer(pod, targetNs)
                },
		agentInjector.InjectAgent,
		metricsAggregator.InjectMetricsAggregator,
	}
```

(but as said elsewhere, I would use a dedicated initalizer for this functionality instead of mangling it into the storage initializer)

Reply:
And, tbh, I don't know why using this complex way of calling for method is used anyways ;-)

Just calling one after each other inline would be much more straight forward in this case. If you would move those list of mutators around method boundaries, then it might make sense, but here it's just overengineered.

I suggest to just use:

```
err := injectGKEAcceleratorSelector(pod);
if err != nil {
   return err
}
err := injectGKEAcceleratorSelector(pod);
if err != nil {
   return err
}
err = storageInitializer.InjectStorageInitializer(pod, targetNs);
if err != nil {
   return err
}
err = agentInjector.InjectAgent(pod);
if err != nil {
   return err
}	
return metricsAggregator.InjectMetricsAggregator;
```

Reply:
I would also question that a method named "injectGKEAccelarator" is a good fit or such a high-level package (while all other injectors are in separate packages, this should be also in a separate package, too)

Reply:
Just realized that this "abstraction" is coming from upstream, so shouldn't be discussed here. So, I conclude for a minimal change I suggest the solution with the closure as this does not require changing the signatures of the other initializers.